drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1121 @@
+ */
+static inline int tmag5273_init_sensor_settings(const struct tmag5273_config *drv_cfg)
+{
+	int retval;
+
+	/* REG_SENSOR_CONFIG_1 */
+	uint8_t regdata = FIELD_PREP(TMAG5273_MAG_CH_EN_MSK, TMAG5273_MAGNETIC_CH_NONE);
+	uint8_t val = ((drv_cfg->x_axis) ? TMAG5273_MAGNETIC_CH_X_EN : TMAG5273_MAGNETIC_CH_NONE) |
+		      ((drv_cfg->y_axis) ? TMAG5273_MAGNETIC_CH_Y_EN : TMAG5273_MAGNETIC_CH_NONE) |
+		      ((drv_cfg->z_axis) ? TMAG5273_MAGNETIC_CH_Z_EN : TMAG5273_MAGNETIC_CH_NONE);
+
+	regdata = TMAG5273_FIELD_SET(regdata, TMAG5273_MAG_CH_EN_MSK, val);
+
+	retval = i2c_reg_write_byte_dt(&drv_cfg->i2c, TMAG5273_REG_SENSOR_CONFIG_1, regdata);
+	if (retval < 0) {
+		LOG_ERR("error setting SENSOR_CONFIG_1");
+		return -EIO;
+	}
+
+	/* REG_SENSOR_CONFIG_2 */
+	regdata = 0;
+
+	val = (drv_cfg->ch_mag_gain_correction == TMAG5273_DT_CORRECTION_CH_2)
+		      ? TMAG5273_MAG_GAIN_CORRECTION_CH_2
+		      : TMAG5273_MAG_GAIN_CORRECTION_CH_1;
+
+	regdata = TMAG5273_FIELD_SET(regdata, TMAG5273_GAIN_CORRECTION_CH_MSK, val);
+
+	switch (drv_cfg->angle_magnitude_axis) {
+	case TMAG5273_DT_ANGLE_MAG_XY:
+		val = TMAG5273_ANGLE_EN_XY;
+		break;
+	case TMAG5273_DT_ANGLE_MAG_YZ:
+		val = TMAG5273_ANGLE_EN_YZ;
+		break;
+	case TMAG5273_DT_ANGLE_MAG_XZ:
+		val = TMAG5273_ANGLE_EN_XZ;
+		break;
+	case TMAG5273_DT_ANGLE_MAG_RUNTIME:
+	case TMAG5273_DT_ANGLE_MAG_NONE:
+		__fallthrough;
+	default:
+		val = TMAG5273_ANGLE_EN_NONE;
+		break;
+	}
+
+	regdata = TMAG5273_FIELD_SET(regdata, TMAG5273_MAG_CH_EN_MSK, val);
+
+	/* measurement range "high" by default */
+	switch (drv_cfg->meas_range) {
+	case TMAG5273_DT_AXIS_RANGE_LOW:
+		regdata = TMAG5273_FIELD_SET(regdata, TMAG5273_X_Y_RANGE_MSK,
+					     TMAG5273_XYZ_MEAS_RANGE_LOW);
+		regdata = TMAG5273_FIELD_SET(regdata, TMAG5273_Z_RANGE_MSK,
+					     TMAG5273_XYZ_MEAS_RANGE_LOW);
+		break;
+	case TMAG5273_DT_AXIS_RANGE_HIGH:
+	case TMAG5273_DT_AXIS_RANGE_RUNTIME:
+		__fallthrough;
+	default:
+		regdata = TMAG5273_FIELD_SET(regdata, TMAG5273_X_Y_RANGE_MSK,
+					     TMAG5273_XYZ_MEAS_RANGE_HIGH);
+		regdata = TMAG5273_FIELD_SET(regdata, TMAG5273_Z_RANGE_MSK,
+					     TMAG5273_XYZ_MEAS_RANGE_HIGH);
+		break;
+	}
+
+	retval = i2c_reg_write_byte_dt(&drv_cfg->i2c, TMAG5273_REG_SENSOR_CONFIG_2, regdata);
+
+	if (retval < 0) {
+		LOG_ERR("error setting SENSOR_CONFIG_2");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief initialize a TMAG5273 sensor
+ *
+ * @param dev handle to the device
+ *
+ * @retval 0 on success
+ * @retval -EINVAL if bus label is invalid
+ * @retval -EIO on communication errors
+ */
+static int tmag5273_init(const struct device *dev)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+	struct tmag5273_data *drv_data = dev->data;
+	int retval;
+
+	if (!i2c_is_ready_dt(&drv_cfg->i2c)) {
+		LOG_ERR("could not get pointer to TMAG5273 I2C device");
+		return -ENODEV;
+	}
+
+	if (drv_cfg->need_int_pin) {
+		/* init interrupt GPIO if present */
+		if (!gpio_is_ready_dt(&drv_cfg->int_gpio)) {
```

Comment:
should probably check that `int_gpio` is not null

Reply:
`int_gpio` is set in `tmag5273_config` via `struct gpio_dt_spec`, so it can't be null here.

Reply:
oh I se what you mean, fine then, you could probably add a `BUILD_ASSERT` to check that the gpio is specified if `trigger-conversion-via-int` so this condition fails at build time if you feel like

Reply:
good idea! Better fail early than at runtime.